src/service/MatchmakingService/MatchController.ts

```diff
@@ -75,6 +75,24 @@ export default class MatchController extends APIClient {
         });
     }
 
+    /**
+     * Fetch matches for a given search profile.
+     * @param searchProfile
+     *      A search profile entity
+     * @param query
+     *      The query to filter special results
+     */
+     async matchForSearchProfileWithBodyAndReportCSV(
+        searchProfile: Entity,
+        query: MatchmakingTypes.FilterQuery = {},
+    ) {
+        return await this.invokeApiWithErrorHandling(`/match/search-profile/csv`, 'POST', searchProfile, {
```

Comment:
Please define return type! 

```suggestion
        return await this.invokeApiWithErrorHandling<???>(`/match/search-profile/csv`, 'POST', searchProfile, {
```

Reply:
Not sure what to do here, the return type is a CSV file, maybe a string?

Reply:
Is it really just the CSV content as string or do you download a file here ?

Reply:
We download the file contents with this endpoint, it's content type is text/csv